addons/hr_expense/models/account_move_line.py

```diff
@@ -3,6 +3,15 @@
 
 from odoo import api, fields, models
 
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def payment_state_matters(self):
+        self.ensure_one()
+        if self.env['hr.expense.sheet'].search([('account_move_id', '=', self.id)]):
```

Comment:
To avoid a search everytime the compute is called leading to a lot of `flush` etc, I suggest `if self.line_ids.expense_id:` instead